filters/numeric/filter.py

```diff
@@ -0,0 +1,67 @@
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from spacy import load
+
+
+def has_number_in_digit(sentence):
+    return any(char.isdigit() for char in sentence)
+
+
+def has_number_in_words(tokens, numbers_in_words):
+    for token in tokens:
+        if token.text.lower() in numbers_in_words:
+            word_flag = True
+            return word_flag
+        elif token.text[-2:] == "th":
+            if token.text[:-2] in numbers_in_words:
+                return True
+    return False
+
+
+class TextContainsNumberFilter(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(self, has_digit=True, has_word=True):
+        super().__init__()
+        self.nlp = load("en_core_web_sm")
+        self.has_digit = has_digit
+        self.has_word = has_word
+        self.numbers_in_words = ["one", "two", "three", "four", "five", "six", "seven", "eight",
```

Comment:
Isn't this an extension of the "keywords" filter?

Reply:
Yes, Kaustubh, you are right. We can see it as an extension of a keywords filter.  But from my understanding, the "Keywords" filter is pretty general in nature and is best suited to filter some special words like "Product name", "Person name" etc (Where filter words are very limited and creating such list is straight forward eg. ["Policy", "Insurance", "Cover"]). On the other hand, in some cases, such lists are relatively long and should be carefully created such that no member word is left out. Now from the NL-aug user's point of view, who need a quick, complete and readymade solution to extract utterances contains numeric data so that he/she can do some future experiments, It could be a useful filter.